### PR TITLE
Add events

### PR DESCRIPTION
**Pull request by clare**
The add events API has been added.   At the moment this is only storing events in the database - there is no api to retrieve them. 

I have added some unit tests and updated the documentation.

### CODE REVIEW

File: api/V1/Event.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+module.exports = function(app, baseUrl) {
+  var apiUrl = baseUrl + '/events';
+
+  /**
+   * @api {post} /api/v1/events Add new events
+   * @apiName PostEvent
+   * @apiGroup Events
+   * @apiDescription This call is used to upload new events.   Events
+   * details are decided by user and can be specified by json or using an
```
Comment (mjs) [inline api/V1/Event.js]: JSON

File: api/V1/Event.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  app.post(
+    apiUrl,
+    [
+      middleware.authenticateDevice,
+      middleware.getEventDetailById.optional(),
+      middleware.isDateArray("dateTimes", "List of times event occured is required."),
+      oneOf([
+        check("eventDetailId").exists(),
+        check("description.type").exists(),
+      ], "Either 'eventDetailId' or 'description.type' must be specified"),
```
Comment (mjs) [inline api/V1/Event.js]: Nice

File: models/EventDetail.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    searchDetails = {
+      $eq: null
+    };
+  }
+
+  return await this.findOne({ where: {
+    type: searchType,
+    details: searchDetails,
+  }});
+};
```
Comment (mjs) [inline models/EventDetail.js]: Indentation appears to be off in `getMatching`. The first level is just one space?

**Comment by clare** [comment]
Add events API so events can be recorded

**Pull request merged**